src/Symfony/Component/Asset/CompilerPass/GitCommitHashCompilerPass.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Asset\CompilerPass;
+
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\Process\Process;
+
+/**
+ * Save git commit hash to parameter and then use it as assets version.
+ *
+ * @author Evgenii Sokolov <ewgraf@gmail.com>
+ */
+class GitCommitHashCompilerPass implements CompilerPassInterface
+{
+    /**
+     * @param ContainerBuilder $container
+     *
+     * @throws \Exception
+     */
+    public function process(ContainerBuilder $container)
+    {
+        $process = new Process('git log -1');
+
+        if (0 !== $process->run()) {
+            throw new \Exception('Git command return wrong exit code');
+        }
+
+        $container->setParameter('git_commit_hash_version_strategy', sha1($process->getOutput()));
```

Comment:
> getting a sha1 of the logs is weird. Why not getting the commit hash instead ?

Because for this parse output needed and result will be same - some unique hash of commit. Less work and no dependency on git output format, I fear that it can be reconfigured.

Why it looks weird? You think have real git commit hash looks better and have advantages?